aiida/orm/implementation/general/calculation/work.py

```diff
@@ -61,8 +61,16 @@ def kill(self):
         """
         Kill a WorkCalculation and all its children.
         """
+        from aiida.orm.calculation.job import JobCalculation
+        from aiida.common.exceptions import InvalidOperation
+
         if not self.is_sealed:
             self._set_attr(self.DO_ABORT_KEY, 'killed by user')
 
         for child in self.get_outputs(link_type=LinkType.CALL):
-            child.kill()
+            try:
+                child.kill()
+            except InvalidOperation as e:
+                if isinstance(child, JobCalculation):
+                    # Cannot kill calculations that are already killed: skip and go to the next step
+                    pass
```

Comment:
Shouldn't we reraise the exception if it is not an instance of `JobCalculation` here? Otherwise the exception is silently swallowed. Not sure if that is the intended behavior